src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util._
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.{SAMFileWriterFactory, SamReaderFactory}
+
+import scala.collection.mutable
+import CorrectUmis._
+
+object CorrectUmis {
+  /**
+    * Holds information about the match of a UMI to an expected UMI.
+    * @param matched true if the match is acceptable, false otherwise
+    * @param umi the fixed UMI sequence that was the closest match
+    * @param mismatches the number of mismatches between the UMI and the reported best matching fixed UMI
+    *  */
+  private[umi] case class UmiMatch(matched: Boolean, umi: String, mismatches: Int)
```

Comment:
I'm not sure that adds anything.  The Umis are pretty much the only strings being flung around, and I think if they were type-def'd I'd spend more time thinking about "wait, are they Strings or byte[]s underneath?"